lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -42,6 +40,7 @@ def fetch_logs
 
     def deploy_succeeded?
       return false unless @latest_rs.present?
+      return partial_deployment_success unless min_unavailable_rollout.blank?
```

Comment:
I think the new concept you're introducing should have first-class visibility here. I.e. I'd move the case statement on `required_rollout_type` (or whatever you name the method) right in here, and have `minimum_unavailable_replicas` assume it's being asked in the context of the relevant rollout strategy.